### DISCUSSION TITLE
Separate Detection rates for VO and for LCD?

### DISCUSSION THREAD

**Discussion by mattiasmar** [introlab/rtabmap#1086] [Q&A] [answered]
Is there an option to specify different detection rates for the visual odometry and the loop closure detection mechanisms?

If the LCD is much more expensive to calculate (due to selection of feature points) then I might want to call it with a lower frequency compared to the VO.

**Comment by matlabbe** [comment, marked as answer]
For loop closure detection, the frame rate is limited by `Rtabmap/DetectionRate` parameter (default 1 Hz).

For visual odometry, frame rate will depend on the input source and processing power. If you feed it 30 Hz images and your computer can process the visual odometry update under 33 ms, you would have 30 Hz odometry. On ros rtabmap's oodmety nodes, you can set `max_update_rate` parameter to throttle input images if you want lower odometry frame rate to save CPU usage. Other option is to throttle input topics externally.
